File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/Items/Triangle.java
package com.spvessel.Items;

import java.util.List;
import com.spvessel.Engine.*;

public class Triangle extends Primitive {
    static int count = 0;
    public int RotationAngle = 32;

    public Triangle() {
        setItemName("Triangle_" + count);
        count++;
    }

    public Triangle(int angle) {
        this();
        RotationAngle = angle;
    }

    @Override
    public List<float[]> makeShape() {
        setTriangles(GraphicsMathService.getTriangle(getWidth(), getHeight(), getX(), getY(), RotationAngle));
        return GraphicsMathService.toGL(updateShape(), getHandler());
    }
}

File: spacevil_21052019/net_src/Items/Triangle.cs
﻿using System.Collections.Generic;
using System.Drawing;
using System.Linq;

namespace SpaceVIL
{
    public class Triangle : Primitive
    {
        static int count = 0;
        public Triangle()
            : base(name: "Triangle_" + count)
        {
            count++;
        }

        public int RotationAngle = 0;
        public override List<float[]> MakeShape()
        {
            SetTriangles(GraphicsMathService.GetTriangle(GetWidth(), GetHeight(), GetX(), GetY(), RotationAngle));
            return GraphicsMathService.ToGL(UpdateShape(), GetHandler());
        }
    }
}